src/coreclr/jit/compiler.hpp

```diff
@@ -2612,32 +2612,19 @@ inline bool Compiler::fgIsThrowHlpBlk(BasicBlock* block)
         return false;
     }
 
-    GenTree* call = block->lastNode();
-
-#ifdef DEBUG
-    if (block->IsLIR())
-    {
-        LIR::Range& blockRange = LIR::AsRange(block);
-        for (LIR::Range::ReverseIterator node = blockRange.rbegin(), end = blockRange.rend(); node != end; ++node)
-        {
-            if (node->OperGet() == GT_CALL)
-            {
-                assert(*node == call);
-                assert(node == blockRange.rbegin());
-                break;
-            }
-        }
-    }
-#endif
+    // Special check blocks will always end in a throw helper call.
+    //
+    GenTree* const call = block->lastNode();
 
-    if (!call || (call->gtOper != GT_CALL))
+    if ((call == nullptr) || (call->gtOper != GT_CALL))
     {
         return false;
     }
 
     if (!((call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_RNGCHKFAIL)) ||
           (call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_THROWDIVZERO)) ||
-          (call->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_THROWNULLREF)) ||
```

Comment:
I guess it should also be removed from comments for `fgIsThrowHlpBlk` function.

Also looks like it's unused in the whole repo and can be removed?

Reply:
At some point I wanted to use it for poor-man's "meaningful" nullreference exceptions via expanding implicit nullchecks to flow-level checks 🙂

Reply:
I don't know the back story of `HELP_THROWNULLREF` -- @erozenfeld do you recall if this was something we added just for LLILC (per https://github.com/dotnet/coreclr/pull/1501)?

Reply:
Yes, I think so.